pyrometheus/__init__.py

```diff
@@ -252,7 +252,7 @@ def troe_falloff_expr(react: ct.Reaction, t):
     troe_params = react.falloff.parameters
     troe_1 = (1.0-troe_params[0])*p.Variable("exp")(-t/troe_params[1])
     troe_2 = troe_params[0]*p.Variable("exp")(-t/troe_params[2])
-    if troe_params[3] > 1.0e-16:
+    if np.abs(troe_params[3]) > 1.0e-16:
```

Comment:
I'll latch onto this, still knowing next to nothing about chemistry, but a bit about floating point. :) It seems that you're testing against machine epsilon. Machine epsilon is an estimate of relative error, using it on things that aren't ratios is likely inherently broken. (The only exception if the denominator in the ratio is O(1) by definition, in which case a comment should explain this.)

Reply:
i can't recall why I simply didn't do `if np.abs(troe_params[3]):`, which would actually make sense, right?

Reply:
Sure, I'd have no objection to that. But you don't even need the `np.abs` then. `if troe_params[3]` should be all you need. Floats are "truthy" when non-zero.

https://stackoverflow.com/questions/39983695/what-is-truthy-and-falsy-how-is-it-different-from-true-and-false

Reply:
That leads me to ask about the original bug...

If the thing you were doing before `if thing > stuff` was a bug that was fixed by changing it to `if np.abs(thing) > stuff`  and now you consider replacing that with just `if thing`, I just can't imagine what the issue is/was.  What was the original issue?  :)

Reply:
Probably `thing` was negative?

Reply:
troe_params[3] appears in some but not all Troe falloff reactions. If present, then the falloff expression is different than if not. Now, I didn't anticipate that it could be _negative_ in some instances where it actually shows up, so I was getting the wrong rate coefficients for those cases. 

In the San Diego mechanism, it only shows up as positive, so our tests didn't detect it. So, as @inducer said, our Troe falloff is *broken*.

Reply:
Now I'm really confused.  If negative is a problem, then how does changing it to `if thing` fix the original bug?  Also if it showed up in SD mech, and it was positive, then how/why was the original (`if thing > eps`) actually a bug?

If I'm just being dense (which is likely), then plz feel free to ignore me.

Reply:
`if troe_params[3]` simply checks that troe_params[3] is non-zero (I think, right @inducer). 

There are mechanisms in which some reactions have a negative `troe_params[3]`, but if we only check that `troe_params[3] > eps`, then it will ignore `troe_params[3] < 0` cases, and produce the wrong expression for Troe falloff.

Reply:
Yup, being dense, but thanks for not ignoring me :)